components/match2/wikis/warcraft/match_summary.lua

```diff
@@ -0,0 +1,347 @@
+	local renderOpponent = function(opponentIx)
+		local opponent = match.opponents[opponentIx]
+		return WarcraftOpponentDisplay.BlockOpponent({
+			flip = opponentIx == 1,
+			opponent = opponent,
+			overflow = 'wrap',
+			showFlag = false,
+			showLink = opponent.type == 'team',
+		})
+			:addClass('brkts-popup-header-opponent')
+	end
+
+	local header = html.create('div'):addClass('brkts-popup-header-dev')
+		:node(renderOpponent(1))
+		:node(renderOpponent(2))
+
+	return header
+end
+
+WarcraftMatchSummary.propTypes.Body = {
+	match = WarcraftMatchGroupUtil.types.Match,
+}
+
+function WarcraftMatchSummary.Body(props)
+	DisplayUtil.assertPropTypes(props, WarcraftMatchSummary.propTypes.Body)
+	local match = props.match
+
+	local body = html.create('div')
+		:addClass('brkts-popup-body')
+		:addClass('brkts-popup-sc-body')
+
+	-- Stream, date, and countdown
+	if match.dateIsExact then
+		local countdownNode = DisplayHelper.MatchCountdownBlock(match)
+			:addClass('brkts-popup-body-element')
+			:addClass('brkts-popup-countdown')
+		body:node(countdownNode)
+	end
+
+	if match.opponentMode == 'uniform' then
+		for _, game in ipairs(match.games) do
+			if game.map or game.winner then
+				body:node(WarcraftMatchSummary.Game(game):addClass('brkts-popup-body-element'))
+			end
+		end
+	else -- match.opponentMode == 'team'
+
+		-- Show the submatch score if any submatch consists of more than one game
+		local showScore = Array.any(match.submatches, function(submatch) return 1 < #submatch.games end)
+
+		for _, submatch in ipairs(match.submatches) do
+			body:node(
+				WarcraftMatchSummary.TeamSubmatch({submatch = submatch, showScore = showScore})
+					:addClass('brkts-popup-body-element')
+			)
+		end
+	end
+
+	-- Vetoes
+	for ix, veto in ipairs(match.vetoes) do
+		local vetoNode = WarcraftMatchSummary.Veto(veto)
+		if ix == 1 then
+			vetoNode = html.create('div')
+				:node(WarcraftMatchSummary.GameHeader({header = 'Vetoes'}))
+				:node(vetoNode)
+		end
+		body:node(vetoNode:addClass('brkts-popup-body-element'))
+	end
+
+	-- Match casters
+	if match.casters then
+		body:node(
+			html.create('div'):addClass('brkts-popup-sc-game-comment')
+				:node('Caster(s): ' .. match.casters)
+		)
+	end
+
+	-- Match comment
+	if match.comment then
+		body:node(
+			html.create('div'):addClass('brkts-popup-sc-game-comment')
+				:node(match.comment)
+		)
+	end
+
+	return body
+end
+
+function WarcraftMatchSummary.Game(game)
+	DisplayUtil.assertPropTypes(game, WarcraftMatchGroupUtil.types.Game.struct)
+
+	local centerNode = html.create('div'):addClass('brkts-popup-sc-game-center')
+		:wikitext(DisplayHelper.MapAndStatus(game))
+
+	local winnerIcon = function(opponentIx)
+		return game.resultType == 'draw' and WarcraftMatchSummary.ColoredIcon('YellowLine')
+			or game.winner == opponentIx and WarcraftMatchSummary.ColoredIcon('GreenCheck')
+			or WarcraftMatchSummary.ColoredIcon('NoCheck')
+	end
+	
+	local heroIcons = function(opponentIx)
+		return game.heroIcons

```

Comment:
this needs to generate the display of the heroIcons
`game.heroIcons` will not be set so this will be just nil